app/controllers/curation_concerns/generic_works_controller.rb

```diff
@@ -1,23 +1,31 @@
 # Generated via
 #  `rails generate curation_concerns:work GenericWork`
-
+require 'edtf'
 class CurationConcerns::GenericWorksController < ApplicationController
   include CurationConcerns::CurationConcernController
   # Adds Sufia behaviors to the controller.
   #Override Sufia behavior to change the after_create message
   include Umrdr::WorksControllerBehavior
 
   before_action :check_recent_uploads, only: [:show]
+  before_action :assign_date_coverage, only: [:create]
   after_action  :notify_rdr, only: [:create]
 
   set_curation_concern_type GenericWork
 
+  def update
+    date_coverage = Umrdr::DateRangeService.new(params).transform
+    params['generic_work']['date_coverage'] = [date_coverage]
```

Comment:
These two lines are the same as the `assign_date_coverage` function.  Why not use the `before_action` as is done for `:create` ?